src/common/MemberSelector/MemberSelector.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useMemo } from "react";
+import { Control, useController } from "react-hook-form";
+import {
+  FormItem,
+  FormLabel,
+  FormControl,
+  FormMessage,
+  FormField,
+} from "@/components/ui/form";
+import {
+  Select,
+  SelectTrigger,
+  SelectValue,
+  SelectContent,
+  SelectItem,
+} from "@/components/ui/select";
+import { UserStatus } from "@/constant/constant";
+import { MemberRole, User } from "@/lib/constants";
+
+interface SelectFieldProps {
+  control: Control<any>;
+  name: string;
+}
+
+const memberOptions = [
+  {
+    id: "1",
+    name: "Alice Johnson",
+    shift: "A",
+    role: "incharge",
+    status: "active",
+  },
+  { id: "2", name: "Bob Smith", shift: "B", role: "member", status: "invited" },
+  {
+    id: "3",
+    name: "Charlie Brown",
+    shift: "C",
+    role: "shift-incharge",
+    status: "active",
+  },
+  {
+    id: "4",
+    name: "Diana Prince",
+    shift: "D",
+    role: "member",
+    status: "active",
+  },
+];
+
+const loginUser: User = {
+  id: 1,
+  name: "Alice Johnson",
+  shift: "A",
+  role: "incharge",
+  status: UserStatus.Active,
+  phone: "123-456-7890",
+  address: "123 Main St, Anytown, USA",
+  created_at: new Date().toISOString(),
+  updated_at: new Date().toISOString(),
+  deleted_at: null,
+};
+
+const MemberSelector = ({ control, name }: SelectFieldProps) => {
+  const { fieldState } = useController({
```

Comment:
How will this component get the list of members when backend is working?

Reply:
Sir @Monfernape  Now I have fixed this by getting member from backend.